codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/codegen/StructureGenerator.java

```diff
@@ -117,134 +113,118 @@ public void renderStructure(Runnable runnable, boolean isInputStructure) {
         writer.write("");
 
         String memberName = symbolProvider.toMemberName(member);
-        writer.writeMemberDocs(model, member);
 
         Symbol memberSymbol = symbolProvider.toSymbol(member);
+
+        var targetShape = model.expectShape(member.getTarget());
+
         if (isInputStructure) {
           memberSymbol =
             memberSymbol
               .getProperty(SymbolUtils.INPUT_VARIANT, Symbol.class)
               .orElse(memberSymbol);
         }
+        var namespace = SmithyNameResolver.smithyTypesNamespace(targetShape);
+
+        if (targetShape.hasTrait(ReferenceTrait.class)) {
+          memberSymbol =
+            memberSymbol.getProperty("Referred", Symbol.class).get();
```

Comment:
(Curious) What does this do? How does it change the symbol?

Reply:
We put a property if a symbol is reference while visiting it. Then when writing all the symbols, we use the replacement.

Ref: https://github.com/smithy-lang/smithy-dafny/blob/ac8a6a47c9e3b7272a90129bc4133ed97e4fac94/codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/codegen/SymbolVisitor.java#L579